osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -201,6 +205,8 @@ public virtual void Dispose()
             waveform.Recycle();
             storyboard.Recycle();
             skin.Recycle();
+
+            beatmapCancellation.Cancel();
```

Comment:
I looked into this a bit and found that `WorkingBeatmap.Dispose()` is never invoked, means we're potentially running multiple loads in the bg.

Reply:
Added finaliser